crates/esplora/src/blocking_ext.rs

```diff
@@ -216,12 +228,12 @@ impl EsploraExt for esplora_client::BlockingClient {
                 }
 
                 let last_index = last_index.expect("Must be set since handles wasn't empty.");
-                let past_gap_limit = if let Some(i) = last_active_index {
+                let gap_limit_reached = if let Some(i) = last_active_index {
                     last_index > i.saturating_add(stop_gap as u32)
```

Comment:
```suggestion
                    last_index >= i.saturating_add(stop_gap as u32)
```

Imagine 10 addresses and we send a tx to the 4th address and the 10th address (like in the test). We need to scan 6 times in order to find the tx at the 10th address (5, 6, 7, 8, 9, 10). So here, when `i` is 3 and the `last_index` scanned is 9, we have

`9 >= 3 + 6`

and we break. 

This means the second half of the test `test_update_tx_graph_gap_limit` should read

`// A scan with gap limit 5 won't find the second transaction, but a scan with gap limit 6 will.`

Reply:
Hey, this was marked as resolved, but the code didn't really change. Unresolving.